src/App.jsx

```diff
@@ -5,21 +5,13 @@ import Map from './components/Map/Map';
 
 function App() {
   const [lastVisit, setLastVisit] = useState('');
-  const [storageCount, setStorageCount] = useState(0);
-  function handleCount() {
-    setTimeout(function () {
-      setStorageCount(storageCount + 1);
-    }, 15000);
-  }
-  handleCount();
   useEffect(() => {
     if (window.localStorage.getItem('lastVisit')) {
       setLastVisit(window.localStorage.getItem('lastVisit').slice(0, 10));
-      console.log(lastVisit);
+    } else {
+      window.localStorage.setItem('lastVisit', new Date().toISOString());
     }
-    window.localStorage.setItem('lastVisit', new Date().toISOString());
-    console.log(window.localStorage.getItem('lastVisit'));
-  }, [storageCount]);
+  }, []);
```

Comment:
J'ai maintenant une erreur quand je charge la page avec un localStorage vierge.
Vérifie la présence de l'entré dans localStorage en premier et si elle n'existe pas, créé-la. Ensuite charge-la tout simplement.

*in a nutshell* : 
```js
if (!window.localStorage.getItem('lastVisit')) {
  window.localStorage.setItem('lastVisit', new Date().toISOString());
}
setLastVisit(window.localStorage.getItem('lastVisit').slice(0, 10));
```

Reply:
Par contre, dans le render
```JS
{lastVisit === '' && (
  <p className="error">This is your first visit, please come again.</p>
)}
```
ne s'exécutera jamais